File: Python/lark_card_kit/models/elements/loop.py
"""Loop element for repeating content."""

from __future__ import annotations

from dataclasses import dataclass, field
from typing import Any

from .base import Element


@dataclass
class Loop(Element):
    """Loop element for repeating template content.

    Repeats a template element for each item in data.
    """

    tag: str = field(default="loop", init=False)
    template: Any = None
    data_key: str | None = None

File: TypeScript/LarkCardKit/src/models/elements/Loop.ts
import type { ElementBase } from './Element';
import type { PlainText } from './PlainText';

export interface Loop extends ElementBase {
  tag: 'loop';
  data_source: LoopDataSource;
  template: ElementBase;
}

export interface LoopDataSource {
  list?: any[];
  [key: string]: any;
}
